docs/examples/metrics/measurement-processors/README.md

```diff
@@ -0,0 +1,245 @@
+# OpenTelemetry Python - MeasurementProcessor Implementation
+
+This implementation adds support for **MeasurementProcessor** to the OpenTelemetry Python SDK, following the [OpenTelemetry Specification PR #4318](https://github.com/open-telemetry/opentelemetry-specification/pull/4318).
+
+## Overview
+
+The MeasurementProcessor allows you to process measurements before they are aggregated and exported. This enables powerful use cases such as:
+
+- **Dynamic injection of additional attributes** to measurements based on Context (e.g., from Baggage)
+- **Dropping attributes** (e.g., removing sensitive information)
+- **Dropping individual measurements** (e.g., filtering invalid values)
+- **Modifying measurements** (e.g., unit conversion, value transformation)
+
+## Key Features
+
+### Chain-of-Responsibility Pattern
+
+Unlike existing processors in OpenTelemetry (SpanProcessor, LogRecordProcessor), MeasurementProcessor uses a **chain-of-responsibility pattern** where each processor is responsible for calling the next processor in the chain. This gives processors fine-grained control over the processing flow.
+
+### High Performance
+
+The implementation is designed for high-performance scenarios:
+- Minimal overhead when no processors are configured
+- Efficient processor chaining using closures
+- No unnecessary object creation in the hot path
+
+## Architecture
+
+```
+Measurement → Processor 1 → Processor 2 → ... → Processor N → Aggregation
+```
+
+Each processor can:
+1. **Pass through unchanged**: `next_processor(measurement)`
+2. **Modify and pass**: `next_processor(modified_measurement)`
+3. **Drop measurement**: Simply don't call `next_processor`
+4. **Split into multiple**: Call `next_processor` multiple times
+
+## Usage
+
+### Basic Setup
+
+```python
+from opentelemetry.sdk.metrics import MeterProvider
+from opentelemetry.sdk.metrics._internal.measurement_processor import (
+    BaggageMeasurementProcessor,
+    StaticAttributeMeasurementProcessor,
+    ValueRangeMeasurementProcessor,
+)
+
+# Create measurement processors
+processors = [
+    BaggageMeasurementProcessor(),  # Add baggage as attributes
+    StaticAttributeMeasurementProcessor({"env": "prod"}),  # Add static attributes
+    ValueRangeMeasurementProcessor(min_value=0),  # Drop negative values
+]
+
+# Configure MeterProvider with processors
+meter_provider = MeterProvider(
+    measurement_processors=processors,
+    # ... other configuration
+)
+```
+
+### Built-in Processors
+
+#### 1. BaggageMeasurementProcessor
+
+Extracts values from OpenTelemetry Baggage and adds them as measurement attributes, enabling end-to-end telemetry correlation.
+
+```python
+# Add all baggage as attributes with "baggage." prefix
+processor = BaggageMeasurementProcessor()
+
+# Add only specific baggage keys
+processor = BaggageMeasurementProcessor(baggage_keys=["user.id", "trace.id"])
```

Comment:
using `trace.id` here is misleading as there is no good use cases for it. To achieve correlation with Distributed Tracing, Exemplars can (should) be leveraged!

Reply:
"user.id" is a good use-case. 

I may also suggest "synthetic_request" as another - its used in [OTel Demo](https://github.com/open-telemetry/opentelemetry-demo/pull/331/files)

(Its not a standard or anything, but something already in demo)